src/main/java/com/commercetools/sync/services/impl/BaseService.java

```diff
@@ -0,0 +1,50 @@
+package com.commercetools.sync.services.impl;
+
+import com.commercetools.sync.commons.BaseSyncOptions;
+import com.commercetools.sync.commons.utils.TriFunction;
+import io.sphere.sdk.commands.DraftBasedCreateCommand;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+class BaseService<U, V> {
```

Comment:
As a discussion:
Don't you want to give some more reasonable abbreviation names for the generics? like:
`BaseService<Res, ResDraft>` 
or
`BaseService<ResType, ResDraftType>`
or as a new convention
`BaseService<ResG, ResDraftG>` (like, use `G` suffix for all generic names)

   Apparently, [_Oracle Java Convention_](https://docs.oracle.com/javase/tutorial/java/generics/types.html) recommends to use single uppercase characters (as you use), but also there are some open discussions on this topic:
https://stackoverflow.com/questions/2900881/generic-type-parameter-naming-convention-for-java-with-multiple-chars
https://dzone.com/articles/naming-conventions
https://stackoverflow.com/questions/6143402/breaking-java-generics-naming-convention

And honestly, i bias for more _readable_ names. This also helps to read the class methods, which accept/return generic types, as same as make usage search between classes/inheritors and so on.

(open a new team discussion on Github, as you recommended)?

Reply:
@andrii-kovalenko-ct I have previously researched this in the beginning and have decided to follow the Oracle convention you have linked. Didn't much think about that it could be confusing. But to be honest, lately I have sometimes found it a bit confusing to know what the generic type refers to in some cases and I had to refer to a place where the generic type is used on an argument for example: @Nonnull final V resourceDraft then I would be able to know that V refers to a type that should be a draft. So I agree with you. But also I am not sure if it would also be too confusing (as Oracle states their argument) that it would be hard to know what is a generic type and what is a Java Object. I know IDEA differentiates it by colour (but is this enough to identify right away what it is) also shall we depend on the IDE that it will differentiate them for all devs of the lib?
Not sure what's the ideal why, maybe it makes sense to get majority opinion from the team, and get different opinions on the topic.

Reply:
could we move the discussion to that new github "flooding" feature?

Reply:
https://github.com/orgs/commercetools/teams/professional-services/discussions/2